examples_utils/benchmarks/command_utils.py

```diff
@@ -183,3 +176,88 @@ def formulate_benchmark_command(
     logger.info(f"new cmd = '{cmd}'")
 
     return cmd
+
+
+def get_poprun_hosts(cmd: list) -> list:
```

Comment:
I think it would be worth starting to write unit tests for each of the functions such as this one - as the code base its getting bigger. It shouldn't be too difficult as its quite self contained. You could separate out the hostname parsing from setting up the hosts functionality

Reply:
Agreed, I want to start adding tests soon before this blows up in size, planning to add a lot of features over the next few weeks. Will integrate pytesting as priority #2 or #3 I think.